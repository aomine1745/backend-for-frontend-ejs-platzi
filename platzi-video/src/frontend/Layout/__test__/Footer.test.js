import React from 'react';
import { create } from 'react-test-renderer';
import { render, shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Footer from '../Footer';

configure({ adapter: new Adapter() });

describe('Footer testing', () => {
  test('Match Snapshot', () => {
    const footer = create(<Footer />);
    expect(footer.toJSON()).toMatchSnapshot();
  });

  test('footer haves class .Footer', () => {
    const footer = shallow(<Footer />);
    const footerElem = footer.find('footer');
    expect(footerElem.hasClass('Footer')).toBe(true);
  });

  test('footer haves 3 anchor tags', () => {
    const footer = render(<Footer />);
    expect(footer.find('a')).toHaveLength(3);
  });
});
