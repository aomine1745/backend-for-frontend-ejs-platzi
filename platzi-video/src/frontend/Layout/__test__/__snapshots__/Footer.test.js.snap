// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer testing Match Snapshot 1`] = `
<footer
  className="Footer"
>
  <ul
    className="Footer__content"
  >
    <li
      className="Footer__item"
    >
      <a
        className="Footer__link"
        href="/"
      >
        Términos de uso
      </a>
    </li>
    <li
      className="Footer__item"
    >
      <a
        className="Footer__link"
        href="/"
      >
        Declaración de privacidad
      </a>
    </li>
    <li
      className="Footer__item"
    >
      <a
        className="Footer__link"
        href="/"
      >
        Centro de ayuda
      </a>
    </li>
  </ul>
</footer>
`;
